src/utilities/aspects.js

```diff
@@ -1,98 +1,162 @@
-          aspect = new Aspect({ aspectKey: type, point1Key: point1, point2Key: point2, orb, orbUsed: maxOrb, language: horoscope._language })
-        } else if (isAspect(point2Data.point, point1Data.point, aspectObject.angle, maxOrb)) {
-          orb = aspectObject.angle - getModuloDifference(point2Data.point, point1Data.point)
-          aspect = new Aspect({ aspectKey: type, point1Key: point1, point2Key: point2, orb, orbUsed: maxOrb, language: horoscope._language })
-        }
-
-        if (!aspect) return
-        if (aspect) {
-          if  (aspect.point1Key === 'northnode' && aspect.point2Key === 'southnode') return // no northnode-southnode aspect
-          if  (aspect.point1Key === 'southnode' && aspect.point2Key === 'northnode') return // no southnode-northnode aspect
-          if  (aspect.point1Key === 'ascendant' && aspect.point2Key === 'midheaven') return // no ascendant-midheaven aspect
-          if  (aspect.point1Key === 'midheaven' && aspect.point2Key === 'ascendant') return // no midheaven-ascendant aspect
-          if (aspects.find(a => a.point1Key === point2 && a.point2Key === point1)) return // No duplicates
-
-          aspects.push(aspect)
-
-          if (Array.isArray(createdaspectTypes[type])) {
-            createdaspectTypes[type].push(aspect)
-          } else {
-            createdaspectTypes[type] = [aspect]
-          }
-
-          if (Array.isArray(createdAspectPoints[point1])) {
-            createdAspectPoints[point1].push(aspect)
-          } else {
-            createdAspectPoints[point1] = [aspect]
+    ...Object.assign({}, ...data.map((aspect) => ({ [aspect.key]: aspect }))),
+  };
+};
+
+export const createAspects = (horoscope) => {
+  const aspectData = getAspectData(horoscope);
+  const aspectTypes = horoscope._aspectTypes;
+  const aspectPoints = horoscope._aspectPoints;
+  const aspectWithPoints = horoscope._aspectWithPoints;
+
+  const aspects = [];
+  const createdaspectTypes = {};
+  const createdAspectPoints = {};
+
+  aspectPoints.forEach((point1) => {
+    aspectTypes.forEach((type) => {
+      aspectWithPoints
+        .filter((p) => p !== point1)
+        .forEach((point2) => {
+          if (!ASPECTS[type])
+            throw new Error(`Could not find data for aspect: "${type}"`);
+          const aspectObject = ASPECTS[type];
+          const maxOrb =
+            horoscope._customOrbs[type] || aspectObject.defaultOrb || 0;
+          const point1Data = aspectData[point1];
+          const point2Data = aspectData[point2];
+
+          let orb, aspect;
+          if (
+            isAspect(
+              point1Data.point,
+              point2Data.point,
+              aspectObject.angle,
+              maxOrb
+            )
+          ) {
+            orb =
+              aspectObject.angle -
+              getModuloDifference(
+                Math.max(point1Data.point, point2Data.point),
+                Math.min(point1Data.point, point2Data.point)
```

Comment:
now takes the larger number first and the smaller number 2nd so the calculation is always correct